types/ember__routing/router-service.d.ts

```diff
@@ -268,6 +268,7 @@ export default class RouterService extends Service {
         modelsD: RouteModel,
         options?: { queryParams: object }
     ): string;
+    urlFor(routeName: string, ...args: RouteModel[] | [...RouteModel[], { queryParams: object }]): string;
```

Comment:
So *theoretically*, this should subsume all the previous overloads, and you should be able to delete them and have the tests keep passing. Does that not actually prove out?

Reply:
That was what I did first, but then wasn't sure whether that would necessitate a major bump so left them all as-is

Reply:
Assuming the existing type tests pass we can remove them: the type tests exist precisely so we can make those kind of type level “refactoring” safely.

Reply:
@nlfurniss I think you should remove the other signatures, and then we're good to go.